provider/cmd/main.go

```diff
@@ -4,18 +4,45 @@
 package main
 
 import (
+	"errors"
 	"fmt"
 	"os"
+	"syscall"
 
 	"github.com/ironcore-dev/libvirt-provider/provider/cmd/app"
+	originzap "go.uber.org/zap"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+const (
+	errTerminatedApp = "app terminated with error"
+)
+
 func main() {
 	ctx := ctrl.SetupSignalHandler()
 
-	if err := app.Command().ExecuteContext(ctx); err != nil {
-		fmt.Println(err.Error())
-		os.Exit(1)
+	cmd, ptrLogger := app.Command()
+	exitCode := int(0)
+	if err := cmd.ExecuteContext(ctx); err != nil {
+		exitCode = 1
+
+		// logger can be nil, if flags weren't parsed
+		if *ptrLogger != nil {
+			(*ptrLogger).Error(errTerminatedApp, originzap.NamedError("error", err))
+		} else {
+			fmt.Fprintln(os.Stderr, errTerminatedApp+": "+err.Error())
+		}
 	}
+
+	if *ptrLogger != nil {
+		err := (*ptrLogger).Sync()
+		// https://github.com/uber-go/zap/issues/991
```

Comment:
This issue is related to running zap under Windows. I can't think of a scenario where we would want to do that in production. In a local dev setup that should be up to the local machine setup and we should not pollute the code base with those workarounds.

Reply:
you have similar issue on mac, this is reason why i added handling of this specific errors.